dds/idl/dds_visitor.cpp

```diff
@@ -309,6 +309,12 @@ dds_visitor::visit_structure(AST_Structure* node)
     field_vec.push_back(*i);
   }
 
+  for (vector<AST_Field*>::iterator it = field_vec.begin(); it != field_vec.end(); ++it) {
+    if (be_global->is_optional(*it)) {
+      idl_global->err()->misc_warning("@optional annotation isn't supported", *it);
```

Comment:
```suggestion
      idl_global->err()->misc_warning("@optional annotation isn't fully supported", *it);
```

We have some support in `DyanmicData`.